src/main/java/seedu/address/model/person/client/Client.java

```diff
@@ -0,0 +1,73 @@
+package seedu.address.model.person.client;
+
+import java.util.Set;
+
+import seedu.address.model.person.Address;
+import seedu.address.model.person.Email;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Phone;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Client in the address book.
+ * Guarantees: name is present and not null, field value is validated, immutable.
+ */
+public class Client extends Person {
+
+    /**
+     * Constructs a client with name given by user and the rest of the fields set to default values.
+     * @param name String representing name of the client
+     */
+    public Client(String name) {
```

Comment:
May not need to have overloaded constructors (see https://github.com/AY2223S1-CS2103-F13-1/tp/pull/62#discussion_r986628503). The default values may be specified in their respective classes, perhaps as static variables?

Reply:
Changing the `Phone`, `Email` and `Address` parameters to optional in `Person` in line with our `client -a` command could also resolve the issue where the parent class `Person` has mandatory parameters while `Client` has optional parameters.

Reply:
> May not need to have overloaded constructors (see [#62 (comment)](https://github.com/AY2223S1-CS2103-F13-1/tp/pull/62#discussion_r986628503)). The default values may be specified in their respective classes, perhaps as static variables?

Noted and edited! Thank you!

Reply:
> Changing the `Phone`, `Email` and `Address` parameters to optional in `Person` in line with our `client -a` command could also resolve the issue where the parent class `Person` has mandatory parameters while `Client` has optional parameters.

Following this implementation, it seems that there are existing tests in the AB3 test suite that fail. Perhaps the static variable quoted above could indicate a non-specified status as opposed to a null status? Alternatively, the EmptyOptional class we had discussed earlier could be used?

Reply:
Yes, we may use the EmptyOptional inner class, with null checks on all parameters as discussed, refer to #62 if unsure of implementation. Do remember to make the static instance accessible and verifiable outside the parameter class (`Phone`, `Email`, `Address`)!

Reply:
> > May not need to have overloaded constructors (see [#62 (comment)](https://github.com/AY2223S1-CS2103-F13-1/tp/pull/62#discussion_r986628503)). The default values may be specified in their respective classes, perhaps as static variables?
> 
> Noted and edited! Thank you!

Wrt this, I wonder if perhaps adding EmptyOptionals inside the other classes of AB3 (name, address, etc.) might somehow affect the logic flow originally implemented? I have made a method to create a default object for these objects with standard values in them (e.g. ANONYMOUS for fullName field under Name) which is used in the client class (therefore, overloading is needed). There is an EmptyOptional for Client class, however. 
Please do feel free to suggest if it is better to have similar classes for name, address etc instead of default values! Ty!

Reply:
I believe there shouldn't be a default static variable for `Client` - the whole purpose of the empty optionals was to provide non-null default values for optional parameters, such as `Email`, `Phone`. Do note that `Address` is not a parameter to `Client`, unlike what I mentioned above and also `Name` is not an optional parameter and hence does not require the default value. Apologies for the confusion.

With regards to the default objects, it might be better to use an inner class, as the default values implemented in this way is accessible by the user. For instance, a user can create a `Client` with `000` as the phone number, although it is very unlikely, meaning that this particular `Client` created by the user may be treated differently from other `Client`(s) since it is meant to be a default value.

On the other hand, the user cannot create an instance of a singleton inner class instance - hence narrowing down the reasons to the default value being encountered, and I cannot see any reason why it would affect the logic flow at the moment.